frontend/context/StakingProgramProvider.tsx

```diff
@@ -68,9 +67,20 @@ const useGetActiveStakingProgramId = () => {
  * It also provides a method to update the active staking program id in state.
  */
 export const StakingProgramProvider = ({ children }: PropsWithChildren) => {
-  const { selectedAgentConfig } = useServices();
+  const {
+    selectedService,
+    selectedAgentConfig,
+    isFetched: isLoaded,
+  } = useServices();
+  const serviceConfigId =
+    isLoaded && selectedService ? selectedService?.service_config_id : '';
+  const { service } = useService({ serviceConfigId });
+
+  // fetch chain data from the selected service
+  const chainId = selectedService?.home_chain_id;
+  const chainData = chainId ? service?.chain_configs[chainId].chain_data : null;
```

Comment:
maybe can just be `serviceId = chainId ? service?.chain_configs[chainId].chain_data?. token : null;`